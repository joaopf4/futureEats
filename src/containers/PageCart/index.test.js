import React from 'react'
import {shallow} from 'enzyme'
import { 
    CartWrapper, 
    Title, 
    DeliveryAddressWrapper,
    AddressTitle,
    PaymentWrapper, 
    DeliveryPrice,
    TotalPrice,
    SubTotalTitle,
    SubTotalPrice,
    PaymentMethod,
    PaymentTitle,
    PaymentRadioGroup,
    RadioPayment,
    ConfirmButton,
    Address,
    RadioCustom } from './style';
import {PageCart} from './index';
import renderer from 'react-test-renderer';



describe ("Testing Cart Page", () => {
    test("Finding subComponents", () => {
        //preparação
        const mockFunction = jest.fn()

        const renderedComponent = shallow(<PageCart setCurrentPage={mockFunction}/>)
        const findCartWrapper = renderedComponent.find(CartWrapper)
        const findTitle = renderedComponent.find(Title) 
        const findDeliveryAdressWrapper = renderedComponent.find(DeliveryAddressWrapper)
        const findAddressTitle = renderedComponent.find(AddressTitle)
        const findPaymentWrapper = renderedComponent.find(PaymentWrapper)
        const findDeliveryPrice = renderedComponent.find(DeliveryPrice)
        const findTotalPrice = renderedComponent.find(TotalPrice)
        const findSubTotalTitle = renderedComponent.find(SubTotalTitle)
        const findSubTotalPrice = renderedComponent.find(SubTotalPrice)
        const findPaymentMethod = renderedComponent.find(PaymentMethod)
        const findPaymentTitle = renderedComponent.find(PaymentTitle)
        const findPaymentRadioGroup = renderedComponent.find(PaymentRadioGroup)
        const findRadioPayment = renderedComponent.find(RadioPayment)
        const findConfirmButton = renderedComponent.find(ConfirmButton)

        //verificação
        expect(findCartWrapper).toHaveLength(1);
        expect(findTitle).toHaveLength(1);
        expect(findDeliveryAdressWrapper).toHaveLength(1);
        expect(findAddressTitle).toHaveLength(1);
        expect(findPaymentWrapper).toHaveLength(1);
        expect(findDeliveryPrice).toHaveLength(1);
        expect(findTotalPrice).toHaveLength(1);
        expect(findSubTotalPrice).toHaveLength(1);
        expect(findSubTotalTitle).toHaveLength(1);
        expect(findPaymentMethod).toHaveLength(1);
        expect(findPaymentTitle).toHaveLength(1);
        expect(findPaymentRadioGroup).toHaveLength(1);
        expect(findRadioPayment).toHaveLength(2);
        expect(findConfirmButton).toHaveLength(1);

    })

    test("handleOnClick", () => {
        //preparação
        const mockOnClick = jest.fn()
        const mockFunction = jest.fn()

        const event = { preventDefault: () => {} };
        const renderedComponent = shallow(
            <PageCart 
            onClick={mockOnClick}
            setCurrentPage={mockFunction}
            >Confirmar
            </PageCart>
            )
        const findButton = renderedComponent.find({name: 'confirmButton'})
        jest.spyOn(event, 'preventDefault');
        //execução
        findButton.simulate("click")
        //verificação
        expect(findButton).toHaveLength(1);
        // expect(event.preventDefault).toBeCalled();
        // expect(mockOnClick.mock.calls.length).toEqual(1);  
      })
      test("Testando styled do CartWrapper", () => {
        const newSnapShot = renderer.create(<CartWrapper />).toJSON();

        expect(newSnapShot).toMatchSnapshot();
    })
    
    test("Testando styled do Title", () => {
        const newSnapShot = renderer.create(<Title />).toJSON();

        expect(newSnapShot).toMatchSnapshot();
    })
    test("Testando styled do DeliveryAddressWrapper", () => {
        const newSnapShot = renderer.create(<DeliveryAddressWrapper />).toJSON();

        expect(newSnapShot).toMatchSnapshot();
    })
    test("Testando styled do AddressTitle", () => {
        const newSnapShot = renderer.create(<AddressTitle />).toJSON();

        expect(newSnapShot).toMatchSnapshot();
    })
    test("Testando styled do Address", () => {
        const newSnapShot = renderer.create(<Address />).toJSON();

        expect(newSnapShot).toMatchSnapshot();
    })
    test("Testando styled do PaymentWrapper", () => {
        const newSnapShot = renderer.create(<PaymentWrapper />).toJSON();

        expect(newSnapShot).toMatchSnapshot();
    })
    test("Testando styled do DeliveryPrice", () => {
        const newSnapShot = renderer.create(<DeliveryPrice />).toJSON();

        expect(newSnapShot).toMatchSnapshot();
    })
    test("Testando styled do TotalPrice", () => {
        const newSnapShot = renderer.create(<TotalPrice />).toJSON();

        expect(newSnapShot).toMatchSnapshot();
    })
    test("Testando styled do PaymentMethod", () => {
        const newSnapShot = renderer.create(<PaymentMethod />).toJSON();

        expect(newSnapShot).toMatchSnapshot();
    })
    test("Testando styled do PaymentTitle", () => {
        const newSnapShot = renderer.create(<PaymentTitle />).toJSON();

        expect(newSnapShot).toMatchSnapshot();
    })
    test("Testando styled do PaymentRadioGroup", () => {
        const newSnapShot = renderer.create(<PaymentRadioGroup />).toJSON();

        expect(newSnapShot).toMatchSnapshot();
    })
    
    test("Testando styled do RadioCustom", () => {
        const newSnapShot = renderer.create(<RadioCustom />).toJSON();

        expect(newSnapShot).toMatchSnapshot();
    })
      
    
})