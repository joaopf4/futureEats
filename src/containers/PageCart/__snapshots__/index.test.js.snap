// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Cart Page Testando styled do Address 1`] = `
<p
  className="MuiTypography-root-1 MuiTypography-body1-10 sc-ifAKCX hoDFhD"
/>
`;

exports[`Testing Cart Page Testando styled do AddressTitle 1`] = `
<p
  className="MuiTypography-root-1 MuiTypography-body1-10 sc-bxivhb lgSpXn"
/>
`;

exports[`Testing Cart Page Testando styled do CartWrapper 1`] = `
<div
  className="sc-bdVaJa iVDUrv"
/>
`;

exports[`Testing Cart Page Testando styled do DeliveryAddressWrapper 1`] = `
<div
  className="sc-htpNat jNKREZ"
/>
`;

exports[`Testing Cart Page Testando styled do DeliveryPrice 1`] = `
<p
  className="MuiTypography-root-1 MuiTypography-body1-10 sc-bZQynM jfRzGN"
/>
`;

exports[`Testing Cart Page Testando styled do PaymentMethod 1`] = `
<div
  className="sc-iwsKbI bSTmat"
/>
`;

exports[`Testing Cart Page Testando styled do PaymentRadioGroup 1`] = `
<div
  className="MuiFormGroup-root-37 sc-gqjmRU bcQBnP"
  role="radiogroup"
/>
`;

exports[`Testing Cart Page Testando styled do PaymentTitle 1`] = `
<p
  className="MuiTypography-root-1 MuiTypography-body1-10 sc-gZMcBi kYPbPP"
/>
`;

exports[`Testing Cart Page Testando styled do PaymentWrapper 1`] = `
<div
  className="sc-EHOje dUFlZL"
/>
`;

exports[`Testing Cart Page Testando styled do RadioCustom 1`] = `
<span
  className="MuiButtonBase-root-54 MuiIconButton-root-48 MuiPrivateSwitchBase-root-44 MuiRadio-root-39 MuiRadio-colorSecondary-43 sc-jTzLTM jQqlbp"
  onBlur={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
>
  <span
    className="MuiIconButton-label-53"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-57"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
    <input
      className="MuiPrivateSwitchBase-input-47"
      onChange={[Function]}
      type="radio"
    />
  </span>
  <span
    className="MuiTouchRipple-root-66"
  />
</span>
`;

exports[`Testing Cart Page Testando styled do Title 1`] = `
<p
  className="MuiTypography-root-1 MuiTypography-body1-10 sc-bwzfXH fMRyqN"
/>
`;

exports[`Testing Cart Page Testando styled do TotalPrice 1`] = `
<div
  className="sc-gzVnrw jlHySg"
/>
`;
